src/InfoBoxes/Content/Engine.cpp

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#include "InfoBoxes/Content/Engine.hpp"
+#include "InfoBoxes/Data.hpp"
+#include "Interface.hpp"
+#include "Units/Units.hpp"
+#include "Formatter/TimeFormatter.hpp"
+#include "Formatter/LocalTimeFormatter.hpp"
+
+#include <tchar.h>
+
+void
+UpdateInfoBoxContentCHT(InfoBoxData &data) noexcept
+{
+  const NMEAInfo &basic = CommonInterface::Basic();
+
+  if (!basic.engine_state.cht_temperature_available.IsValid()) {
+    data.SetInvalid();
+    return;
+  }
+
+  data.SetValue(_T("%3.0f"), basic.engine_state.cht_temperature.ToUser());
+  data.SetValueUnit(Units::current.temperature_unit);
+}
+
+void
+UpdateInfoBoxContentETG(InfoBoxData &data) noexcept
+{
+  const NMEAInfo &basic = CommonInterface::Basic();
+  if (!basic.engine_state.egt_temperature_available.IsValid()) {
+    data.SetInvalid();
+    return;
+  }
+  data.SetValue(_T("%3.0f"), basic.engine_state.egt_temperature.ToUser());
+  data.SetValueUnit(Units::current.temperature_unit);
+}
+
+void
+InfoBoxContentRPM::Update(InfoBoxData &data) noexcept
+{
+  const NMEAInfo &basic = CommonInterface::Basic();
+  if (!basic.engine_state.revs_per_sec_available.IsValid()) {
+    data.SetInvalid();
+    return;
+  }
+  // display as revolutions per minute / 1000, so that for instance 8400 RPM are shown as the more compact 8.40
+  data.SetValue(_T("%1.2f"), basic.engine_state.revs_per_sec * (60.0/1000.0));
```

Comment:
Unit is missing, and dividing by 1000 seems rather unintuitive

Reply:
Unit would be 1/(60 s) and with common scaling to 1000th of RPMs, the look of the unit gets worse. You insist on a unit for RPMs?

Reply:
I have not much say in this and will probably not use it anytime soon, but I would also prefer "RPM". Are there digital displays that show RPM as a fraction? I know analogue ones do for obvious reasons, but that does not apply here.
Anyway, we show larger values (for example RadioFrequency) in the infoboxes, so 4-5 digits for engine RPM does not sound excessive to me at least.

Reply:
#1193

Reply:
> So "RPM" as unit in the InfoBox? Do I have to extend `class Units`?

In the long run, we should do that, but for now, I'd agree to merge something that sets "rpm" as "Comment". Not quite consistent, but barely good enough to get it merged.